server/etcdserver/api/etcdhttp/health.go

```diff
@@ -82,8 +90,10 @@ func NewHealthHandler(lg *zap.Logger, hfunc func(ctx context.Context, excludedAl
 		defer func() {
 			if h.Health == "true" {
 				healthSuccess.Inc()
+				recordMetrics(checkTypeHealth, "root", HealthStatusSuccess)
```

Comment:
I'm not sure about support of old health endpoint.

/health endpoint supports parameters that influence the results. This can mean that if there is 2 probers using different parameters (one using health for liveness, second for readiness) it will lead to metric flipping between 0 and 1 without any reason. This is the flaw of the old health probe that we shouldn't inherit.